File: Arrays/06_Union of two arrays/Solution.py
'''
 *  Name - Jatin Sharma
 *  Date - Oct, 28, 2020
 *  Link - https://practice.geeksforgeeks.org/problems/union-of-two-arrays/0
 '''

for _ in range(int(input())):
    n, m = (int(i) for i in input().split())
    arr1 = list(map(int, input().strip().split()))
    arr2 = list(map(int, input().strip().split()))
    print(len(set(arr1 + arr2)))

File: Arrays/06_Union of two arrays/Solution.cpp
/*
 *  Name - Jatin Sharma
 *  Date - Oct, 28, 2020
 *  Link - https://practice.geeksforgeeks.org/problems/union-of-two-arrays/0
 */

#include <iostream>
#include <unordered_set>

using namespace std;

int main()
{
    int testCases;
    cin >> testCases;
    int m, n, s;

    while (testCases--)
    {
        cin >> m >> n;
        unordered_set<int> set;

        for (size_t i = 0; i < m + n; i++)
        {
            cin >> s;
            set.insert(s);
        }

        cout << (int)set.size() << endl;
    }
    return 0;
}